### ISSUE TITLE
Removal of `this` context in newer versions

### ISSUE THREAD

**Issue by christiansaiki** [TheBrainFamily/cypress-cucumber-preprocessor#96]
**TL;DR**
So to summarize I've got two questions:

1. Why `this` context was removed in the newer versions?

1. What is the recommended way to pass context to other tests? Please see the snippet below.

=====================================================================

Hey guys, I'm loving your library and it makes testing much more pleasing and understandable.

In my company we were using version 1.0.0 and for some reason when I upgraded the preprocessor to version 1.4.0 our tests broke completely. We kept getting `Cannot set property targetPage of undefined` errors.

Here is a quick snippet

```javascript
const { Login } = require('./../pages');

given(/^the SIP Demo is loaded$/, function() {
  this.targetPage = new Login().open();
});

when(/^I choose Proof of Identity in the dropdown$/, function() {
  this.targetPage.selectProofOfIdentity();
  this.targetPage.loginButton.should('exist');
});

then(/^a QR Code is displayed$/, function() {
  this.targetPage.qrCodeIframe.should('exist');
});
```
Thank you so much!
Cheers

**Comment by lgandecki** [comment]
that is interesting.. I wouldn't think that this actually ever worked.. But maybe it did, just without having the mocha context on it. Anyway, please try 1.5.0 and let me know if that helps.

**Comment by lgandecki** [comment]
You can also share the context like this: 
https://github.com/TheBrainFamily/cypress-cucumber-preprocessor#sharing-context

but you shouldn't be dissalowed to use "this". :-)

**Comment by christiansaiki** [comment]
Thanks, @lgandecki, yeah when I downgraded to 1.0.0 the tests worked flawlessly.
I guess you didn't publish the new version to npm, I couldn't find it yet.
I will test it once you guys upgrade in npm.

Thanks a lot.

**Comment by christiansaiki** [comment]
Worked flawlessly!
Thank you so much @lgandecki!
